apps/dsp-app/src/app/project/ontology/resource-class-info/add-property-menu.component.ts

```diff
@@ -135,36 +136,36 @@ export class AddPropertyMenuComponent {
 
   trackByPropFn = (index: number, item: PropertyInfoObject) => `${index}-${item.propDef?.id}`;
 
-  assignNewProperty(prop: PropertyInfoObject) {
-    const propertyAssignment: PropertyAssignment = {
-      resClass: this.resourceClass,
-      property: {
-        propType: prop.propType,
-        propDef: prop.propDef,
-      },
-    };
-
+  assignExistingProperty(prop: PropertyInfoObject) {
     const maxGuiOrderProperty = this.resourceClass.propertiesList.reduce(
       (prev, current) => Math.max(prev, current.guiOrder ?? 0),
       0
     );
 
-    const ontology = this._store.selectSnapshot(OntologiesSelectors.currentOntology);
-    this._dialog
-      .open<AssignPropertyDialogComponent, AssignPropertyDialogProps>(
-        AssignPropertyDialogComponent,
-        DspDialogConfig.mediumDialog({
-          ontologyId: ontology.id,
-          lastModificationDate: ontology.lastModificationDate,
-          propertyInfo: propertyAssignment.property,
-          resClassIri: this.resourceClass.id,
-          maxGuiOrderProperty,
-        })
-      )
-      .afterClosed()
-      .pipe(filter(res => res === true))
-      .subscribe(res => {
-        this.updatePropertyAssignment.emit(ontology.id);
+    const currentOntology = this._store.selectSnapshot(OntologiesSelectors.currentOntology);
+
+    const onto = new UpdateOntology<UpdateResourceClassCardinality>();
+    onto.lastModificationDate = currentOntology.lastModificationDate;
+    onto.id = currentOntology.id;
+
+    const addCard = new UpdateResourceClassCardinality();
+    addCard.id = this.resourceClass.id;
+
+    addCard.cardinalities = [
+      {
+        propertyIndex: prop.propDef.id,
+        cardinality: 1,
+        guiOrder: maxGuiOrderProperty + 1,
+      },
+    ];
```

Comment:
is cardinality: 1 good ?

Reply:
Yup, simple default which has no side effects whatsoever.
/**
     * Cardinality 0-1 (optional).
     */
    "_0_1" = 1,